rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/metrics/HttpClientMetricsService.java

```diff
@@ -69,19 +69,21 @@ public void update(String uri, String statusDescription, long durationMs) {
      * @param cause Throwable to unwrap
      * @return string description
      */
-
     public static String unwrapExceptionString(Throwable cause) {
-        // Unwrap the failure and return message
-        if (cause instanceof Http.Failure) {
+        // HttpStatusException is a sub-type of Http.Failure: check it first.
+        if (cause instanceof Http.HttpStatusException) {
+            return String.valueOf(((Http.HttpStatusException)cause).getCode());
+        } else if (cause instanceof Http.Failure) {
             final Throwable rootCause = cause.getCause();
-            if (rootCause instanceof EOFException) {
-                return rootCause.getClass().getName();
+            if (rootCause != null) {
+                if (rootCause instanceof EOFException) {
+                    // message is unique for each exception, group them by name
+                    return rootCause.getClass().getName();
+                }
+                return rootCause.toString();
             }
-            return rootCause.toString();
-        } else if (cause instanceof Http.HttpStatusException) {
-            return String.valueOf(((Http.HttpStatusException)cause).getCode());
         }
-        return "exception";
+        return cause.toString();
```

Comment:
As a compromise, cause.getClass().toString(), but I don't really expect a lot of different exceptions.